### DISCUSSION TITLE
How to transfer bash completion rules

### DISCUSSION THREAD

**Discussion by FranzAtGithub** [xonsh/xonsh#4597] [Q&A] [answered]
Hi,
is there an easy way to reuse the completion rules that I have setup for my bash?

**Comment by gforsyth** [comment]
Can you clarify what you mean by completion rules?

**Comment by FranzAtGithub** [comment]
In my bashrc I have lines like these
`complete -o plusdirs -f -X '!*.py' python3`
that make my TAB completion only consider directories or files ending with ".py" as candidates for a `python3` command.
I source these files with `source_bash` but it seems to have no effect.

**Reply by gforsyth** [reply]
Gotcha.
Put all of your `complete` rules in a separate file, then add that file to `$BASH_COMPLETIONS` and they should be respected

**Reply by FranzAtGithub** [reply]
got it. Is this the usual way for bash as well?

**Reply by gforsyth** [reply]
I'm not sure what the best way to do this is with `bash` itself, but we handle completions separately from other variables and functions defined in bash files

**Reply by FranzAtGithub** [reply]
Nvm. This works for me. Do you want to make your comment an answer actual answer?

**Comment by gforsyth** [comment, marked as answer]
If you put all of your complete rules in a separate file, then add that file to `$BASH_COMPLETIONS` they should be respected
